src/main/java/seedu/address/logic/commands/SetCommand.java

```diff
@@ -0,0 +1,142 @@
+    public static final String MESSAGE_NO_CHANGE = "Ingredient level already set.";
+    private final Ingredient target;
+    private final SetIngredientDescriptor setIngredientDescriptor;
+
+    /**
+     * Constructs a set command with the given ingredient name and amount.
+     */
+    public SetCommand(Ingredient target, SetIngredientDescriptor setIngredientDescriptor) {
+        requireNonNull(target);
+        requireNonNull(setIngredientDescriptor);
+
+        this.target = target;
+        this.setIngredientDescriptor = setIngredientDescriptor;
+
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Ingredient> lastShownList = model.getFilteredIngredientList();
+
+        Ingredient ingredientToEdit = lastShownList.get(lastShownList.indexOf(target));
+        Ingredient updatedIngredient = createSetIngredient(target, setIngredientDescriptor);
+
+        if (!ingredientToEdit.isSameIngredient(updatedIngredient)
+                && model.hasIngredient(updatedIngredient)) {
+            throw new CommandException(MESSAGE_NO_CHANGE);
+        }
+
+        model.setIngredient(ingredientToEdit, updatedIngredient);
+        return new CommandResult(String.format(MESSAGE_SUCCESS));
+    }
+
+    private static Ingredient createSetIngredient(Ingredient target,
+                                                  SetIngredientDescriptor setIngredientDescriptor) {
+        assert target != null;
+
+        Amount updatedAmount = setIngredientDescriptor.getAmount().orElse(target.getAmount());
+
+        return new Ingredient(target.getIngredientName(), updatedAmount);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+        if (!(other instanceof SetCommand)) {
+            return false;
+        }
+
+        SetCommand e = (SetCommand) other;
+
+        return target.equals(e.target)
+                && setIngredientDescriptor == e.setIngredientDescriptor;
+    }
+
+    public static class SetIngredientDescriptor {
+
+        private IngredientName ingredientName;
+        private Amount amount;
+
+        public SetIngredientDescriptor(){}
+
+        /**
+         * Copy constructor.
+         */
+        public SetIngredientDescriptor(SetIngredientDescriptor toCopy) {
+            setAmount(toCopy.amount);
+        }
+
+        /**
+         * Returns true if at least one field is edited.
+         */
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyNonNull(amount);
+        }
+
+        public void setAmount(Amount amount) {
+            this.amount = amount;
+        }
+        public Optional<IngredientName> getIngredientName() {
+            return Optional.ofNullable(ingredientName);
+        }
+        public Optional<Amount> getAmount() {
+            return Optional.ofNullable(amount);
+        }
```

Comment:
You are missing empty lines in between the methods.

Reply:
Noted, will add in.